### ISSUE TITLE
CI: Unpin openpyxl

### ISSUE THREAD

**Issue by TomAugspurger** [pandas-dev/pandas#22605]
Followup to https://github.com/pandas-dev/pandas/pull/22601 and https://github.com/dask/dask-ml/issues/84#issuecomment-418640761.

Upstream issue: https://bitbucket.org/openpyxl/openpyxl/issues/1094/error-with-pandas-datetimeindex

(one of several in fact filed for this issue)

**Comment by gfyoung** [comment]
Upstream commit just came in to patch this:

https://bitbucket.org/openpyxl/openpyxl/commits/fa13fbfda13a

Need to wait on next release to unpin this.

**Comment by h-vetinari** [comment]
Reported error in https://bitbucket.org/openpyxl/openpyxl/issues/1094/error-with-pandas-datetimeindex (thru being duplicate of https://bitbucket.org/openpyxl/openpyxl/issues/1093/newly-introduced-keyerror-problem-in) was fixed in 2.5.7 apparently - current version is 2.5.12. Tagging @alimcmaster1, as he asked about just that on the bitbucket tracker.

**Comment by alimcmaster1** [comment]
@h-vetinari . I already largely did this #23361 the only build that's that remain pinned are azure-macOS-35, and azure-27-compat build. So I think this is perhaps fine to close unless anyone thinks there is something else we should be doing here?

**Comment by gfyoung** [comment]
@jreback @TomAugspurger : Do we intend to unpin `openpyxl` for these two builds?

Otherwise, I do agree that we should close this (and mark for `0.24.0` instead of `0.24.1` in fact).

**Comment by jreback** [comment]
are we pinning anywhere?

**Comment by TomAugspurger** [comment]
```
ci/deps/travis-27.yaml
21:  - openpyxl=2.4.0

ci/deps/azure-27-compat.yaml
11:  - openpyxl=2.5.5

ci/deps/azure-macos-35.yaml
15:  - openpyxl=2.5.5

ci/deps/azure-27-locale.yaml
11:  - openpyxl=2.4.0
```

We have builds that are not pinned though, so this is probably fine to close.

**Comment by TomAugspurger** [comment]
Closing, since we have a mix of pinned & unpinned.
